src/js/components/Breadcrumb.js

```diff
@@ -0,0 +1,295 @@
+   * params. The returned label is the text shown for the crumb. The route is
+   * the route name which will be set on the Link-to attribute. The params will
+   * also be set on the Link tag. Each returned crumb will be spliced into the
+   * resulting breadcrumb.
+   */
+  buildCrumb(label, route, routeParams, labelIsParam) {
+    // Split label on whitespace and -
+    let splitLabelRegex = /[\s-]+/;
+    let joinLabelCharacter = ' ';
+    let params = {};
+
+    if (labelIsParam) {
+      // Extract param value as label e.g. (:userID -> Foo)
+      // and add param to params.
+      //
+      // Remove colon first
+      let paramName = label.slice(1);
+      // Decode value
+      label = decodeURIComponent(routeParams[paramName]);
+      params[paramName] = label;
+    } else {
+      // Split Label on space and - then capitalize each word and join.
+      // /foo-bar/ --> Foo Bar
+      label = label.split(splitLabelRegex).map(function (word) {
+        return StringUtil.capitalize(word);
+      }).join(joinLabelCharacter);
+    }
+    // Can return multiple route objects to get rendered in crumbs.
+    // Useful for splitting a route parameter like a file path.
+    return [{
+      label,
+      route,
+      params
+    }];
+  }
+
+  renderCrumbsFromRoute() {
+    let {router} = this.context;
+    let {buildCrumb, shift} = this.props;
+
+    if (buildCrumb == null) {
+      buildCrumb = this.buildCrumb;
+    }
+
+    let crumbKey = 1;
+    let namedRoutes = this.getCurrentNamedRoutes();
+    let params = router.getCurrentParams();
+    // Remove n items from beginning
+    let paths = Object.keys(namedRoutes).slice(shift);
+
+    return paths.reduce((crumbs, path, pathIndex) => {
+      let name = namedRoutes[path];
+      let pathParts = path.split('/');
+      // Decode Label
+      let label = decodeURIComponent(pathParts[pathParts.length - 1]);
+      let labelIsParam = label.charAt(0) === ':' && label.slice(1) in params;
+
+      let crumbObjects = buildCrumb(
+        label,
+        name,
+        params,
+        labelIsParam
+      );
+
+      crumbObjects.forEach((crumbParams, crumbIndex) => {
+        let {label, route, params} = crumbParams;
+
+        crumbs.push(
+          <li key={crumbKey++}>
+            <Link to={route}
+                params={params}
+                title={label}>
+              {label}
+            </Link>
+          </li>
+        );
+        // Push divider ( > )
+        if (pathIndex !== paths.length - 1
+          || crumbIndex !== crumbObjects.length - 1) {
+
+          crumbs.push(
+            <li key={crumbKey++} >
+              <IconChevron
+                className="icon icon-small"
+                isForward={true} />
+            </li>
+          );
+        }
+      });
+
+      return crumbs;
+    }, []);
+  }
+
+  render() {
+    let classSet = classNames(
+      'list-unstyled breadcrumb',
+      {
+        collapsed: this.state.collapsed,
+        inverse: this.props.inverse
```

Comment:
Could we just add the inverse class and use `classnames/dedupe` to let the user override it if needed, like we are doing here?
https://github.com/dcos/dcos-ui/pull/214/files#diff-6d467002be6b32378982c60715cf6919R110